ramips: Improve support for Phicomm K2P (and KE2P)

rogerpueyo opened the pull request:
This series of commits improve the support for the Phicomm K2P devices:

1. add kmod-mt7615e to the images
Now that the mt76 driver supports the MT7615D chip, found on the Phicomm K2Pdevices, it should be included by default. This commit adds it, as well as the wpad-basic package.
The driver supports operation on both the 2.4 GHz and the 5 GHz bands already, although not yet concurrently (only one band can be used at a time right now).

2. rename K2P to K2P/KE2P
The manufacturer sells the two models, K2P and KE2P, which have identical hardware but are targeted to different markets (the latter being for the European market). Adding the KE2P reference makes things more clear.

adrianschmutzler commented on target/linux/ramips/base-files/lib/ramips.sh:
```diff
@@ -256,7 +256,7 @@ ramips_board_detect() {
 		name="jhr-n926r"
 		;;
-	*"K2P")
+	*"K2P/KE2P")
```
Will the content of /proc/cpuinfo actually change or do you require
```
*"K2P"|\
*"KE2P")
```

dengqf6 replied:
Or just drop this case and use generic board detection instead.

adrianschmutzler replied:
Well, you cannot simply drop both, as then the DTS compatible will be used, which in return does not match the Device definition name in Makefile. Since this is all mixed in ramips target anyway, I've recently started a PR to harmonize (https://github.com/openwrt/openwrt/pull/2171) the entire target, which will then also remove ramips.sh completely.

For the current situation, you just have to check what `awk 'BEGIN{FS="[ \t]+:[ \t]"} /machine/ {print $2}' /proc/cpuinfo` returns on you KE2P. If it ends with K2P, leave this unchanged. If it ends with K2EP, stick with my suggestion for the moment.

981213 replied:
@adrianschmutzler The machine string in /proc/cpuinfo is the model string defined in dts. This commit changed that string to "Phicomm K2P/KE2P" so his approach here is correct.

adrianschmutzler replied:
Ah, I was not aware of that, I thought this was coming from manufacturer or similar.
Sorry for the disturbance.

ynezz commented on target/linux/ramips/dts/K2P.dts:
```diff
@@ -7,7 +7,7 @@
 
 / {
 	compatible = "phicomm,k2p", "mediatek,mt7621-soc";
```
What about the `compatible`, shouldn't you add one for ke2p as well?

ynezz commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -231,7 +231,7 @@ TARGET_DEVICES += iodata_wn-gx300gr
 define Device/k2p
```
[As in](https://github.com/openwrt/openwrt/pull/2138#discussion_r299796314) I'm wondering how the potential user is going to find out, that image for k2p also supports ke2p device as the firmware image naming is unchanged, thus there will be only k2p in the firmware filename.

adrianschmutzler replied:
IMO, if one really wants to achieve that, one would have to add a device definition for ke2p (just copied from k2p) and duplicate lines elsewhere (01_leds, 02_network, ...) as well as DTS.
This will cause another image having to be built, but still I consider it the tidiest solution.
(I personally dislike merging both names into the device ID, e.g. phicomm_k2p-ke2p).

Nevertheless, this is a topic coming up regularly. I have a similar problem with the Ubiquiti XM devices (Bullet M, Loco M, Picostation M, ...): ubnt_bullet-m image is used for all of them, but I'd like to distinguish as all of them have different antenna_gain I need to set.

adrianschmutzler commented:
There has been a reorganization in ramips target today.
This has removed ramips.sh and will use the device compatible instead.
Your first patch will just need a rebase, for the second patch this will slightly touch discussion if you want to change device compatible. If you just want to change DEVICE_MODEL, it would be a rebase, too. However, this should be discussed further, as it is a general problem.

981213 commented:
merged&rebased the first commit in this PR.

iscilyas commented:
i don't know how else to get (any) attention for my PR, but i'm trying to add the kmod-mt7615e packages to the xiaomi MIR3P ... https://github.com/openwrt/openwrt/pull/2293

if the people who are reviewing this PR could also take a look at that... it's been over a week with absolutely no activity (and it's really not controversial)..

adrianschmutzler commented:
This might also be a possible use-case for the DEVICE_ALT0_MODEL syntax: #2250

981213 commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -468,7 +468,7 @@ define Device/phicomm_k2p
   IMAGE_SIZE := 15744k
   DEVICE_VENDOR := Phicomm
-  DEVICE_MODEL := K2P
+  DEVICE_MODEL := K2P/KE2P
```
Hi! You could update to the recently added ALT0_MODEL here now :)

adrianschmutzler replied:
@rogerpueyo Just put:
DEVICE_VENDOR := Phicomm
DEVICE_MODEL := K2P
DEVICE_ALT0_VENDOR := Phicomm
DEVICE_ALT0_MODEL := KE2P

rogerpueyo replied:
I was pushing to the wrong branch. Here it is :)

rogerpueyo committed:
ramips: mt7621: Alternative name KE 2P for Phicomm K2P

The Phicomm KE 2P is identical to the already supported Phicomm K2P,
renamed for the European market. Use the ALT0 buildroot tags to show
both devices.

Signed-off-by: Roger Pueyo Centelles <roger.pueyo@guifi.net>

981213 commented:
Merged into my staging tree at: https://git.openwrt.org/?p=openwrt/staging/981213.git
Thanks!

ewangsoft commented:
My device is  Phicomm K2P,version A1.
firmware:OpenWrt SNAPSHOT r12570-f68c48f / LuCI Master git-20.074.84698-03af89a
When wireless and wired interfaces are bridged together,the wireless network will not work.
Can you help me, thank you！
